### DISCUSSION TITLE
code generation with Knitro and CasADi

### DISCUSSION THREAD

**Discussion by mfocchi** [casadi/casadi#3376] [Q&A]
Dear all 
we need to generate cpp code with knitro and Casadi to solve an nlp in python 
in the specific we are following this example replacing "ipopt" with "knitro"
https://github.com/casadi/casadi/blob/main/docs/examples/python/nlp_codegen.py

it works both with both ipopt and knitro if we set "jit" option
however, we would like to set "external" option and use knitro. In this case it spits an error: "which_depends not defined for External"

Note, that we already solved all the known issues to use  knitro (export LD_PRELOAD=knitrofolder/lib/libiomp5.so)

does anybody have any clue?

**Comment by jgillis** [comment]
Pass option detect_linear_constraints False
